# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/GLImages.h
# Commit: da24abf34123
# Full Hash: da24abf341236e00fa47e9e55dc72c82ae8a8c84
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-12-14 17:20:20
# Regressor Bug: 1866020
# File Overlap Count: 1
# Description:
#   Bug 1866020 - Override buggy colorspace conversion on Pixel devices. r=gw,padenot,geckoview-reviewers,owlish
#   
#   Pixel 6, 7, and 8 devices running Android 14 are affected by a bug
#   where video frames with SMPTE 432 color primaries are rendered
#   incorrectly when sampled from an external texture in GLES. To work
# ==============================================================================

diff -r 3a34e7a424e0 -r da24abf34123 gfx/layers/GLImages.h
--- a/gfx/layers/GLImages.h	Thu Dec 14 10:08:52 2023 +0000
+++ b/gfx/layers/GLImages.h	Thu Dec 14 10:45:08 2023 +0000
@@ -54,6 +54,7 @@
   SurfaceTextureImage(AndroidSurfaceTextureHandle aHandle,
                       const gfx::IntSize& aSize, bool aContinuous,
                       gl::OriginPos aOriginPos, bool aHasAlpha,
+                      bool aForceBT709ColorSpace,
                       Maybe<gfx::Matrix4x4> aTransformOverride);
 
   gfx::IntSize GetSize() const override { return mSize; }
@@ -61,6 +62,7 @@
   bool GetContinuous() const { return mContinuous; }
   gl::OriginPos GetOriginPos() const { return mOriginPos; }
   bool GetHasAlpha() const { return mHasAlpha; }
+  bool GetForceBT709ColorSpace() const { return mForceBT709ColorSpace; }
   const Maybe<gfx::Matrix4x4>& GetTransformOverride() const {
     return mTransformOverride;
   }
@@ -102,6 +104,7 @@
   bool mContinuous;
   gl::OriginPos mOriginPos;
   const bool mHasAlpha;
+  const bool mForceBT709ColorSpace;
   const Maybe<gfx::Matrix4x4> mTransformOverride;
   UniquePtr<SetCurrentCallback> mSetCurrentCallback;
 };